app/main.py

```diff
@@ -1,25 +1,86 @@
 from abc import ABC
+from typing import Any, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: Any, owner: Type) -> int:
+        return getattr(instance, self.name)
+
+    def __set__(self, instance: Any, value: int) -> None:
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"{self.name} must be between \
+                {self.min_amount} and {self.max_amount}")
+        setattr(instance, self.name, value)
+
+    def __set_name__(self, owner: Type, name: str) -> None:
+        self.name = name
```

Comment:
Add an underscore before name to define the protected name:
```suggestion
        self.protected_name = "_" + name
```